### DISCUSSION TITLE
Create one dicom file from multi jpeg file in c#.

### DISCUSSION THREAD

**Discussion by Dash-Uttam** [fo-dicom/fo-dicom#1225] [Q&A] [answered]
Hello All,

I need to create one single dicom file from multiple jpeg files in c#. If can anyone has done please provide guidance or reference code ?

Thanks in Advance.

**Comment by mia-san** [comment, marked as answer]
hoping this helps.
https://gist.github.com/mia-san/7d5b4a333b36d854901c2b180f1e8950

you might need some changes if you're using version 5.
//

**Comment by Tjossul** [comment]
Hello,

I would like to expand the question...

I have 100 images with about 30 MB (4096 x 3000 pixels) each that I want to add to the DICOM. If I open the created 3.7 GB DICOM with a (free to use) viewer the  59th frame and following are destroyed (RGB banner only if I would describe them). 

Did you test creating large DICOMs? If yes, do you have a sample that surely produces a DICOM with intact frames that I can work with?

Thanks in advance!
T.

**Reply by mia-san** [reply]
No, I think I haven't created such a huge DICOM file.
well, are you using ver 5?
have you checked if other program works?
e.g. dcmtk or dcm4che seems to have some programs to manipulate and investigate DICOM file.

**Reply by Tjossul** [reply]
Hello mia-san,

I'm using fo-dicom v5.1.4 and fo-dicom.NetCore v4.0.8.

I didn't try the programs you mentioned but I will do.

I come back later with answers.

Thanks for now...
T.

**Reply by Tjossul** [reply]
Hello,

I tried some viewer applications such as microDicom, RadiAnt that opens the DICOM (described above) with the same problems. 

The DCMTK related DICOMscope does not open the file and ClearCanvas Viewer which is based on dcm4chee also did not open the file. 

The Weasis Viewer is based on dcm4che and opens the file with the same problems I described above.

Could you please try to reproduce this? It is simply adding 100 big RGB images to a DICOM and open it with a viewer.

Thanks in advance!
T.

**Reply by mia-san** [reply]
hmm, then produced DICOM file is bad.

what I can imagine is
- are you using fo-dicom.NetCore v4.0.8 with fo-dicom v5.1.4 ? if it is, that sounds strange to me.
- does transfer syntax matters, such as JPEG Baseline or Uncompressed ?
- what happens if adding same image 100 times?
- what happens if adding samller images that exceeds 2GB or so?

otherwise I have no idea.

I don't have "100 big RGB images" myself, so I don't know if I could reproduce the issue.

**Reply by Tjossul** [reply]
I removed the fo-dicom.NetCore v4.0.8 (which is the latest version) and use now FellowOakDicom instead. 

This works!

Thanks for your suggestions.

**Reply by mia-san** [reply]
no problem.
glad to know it worked :)

note that fo-dicom ver 5 officially "Targets .NET Standard 2.0", so you don't need ver 4 components anymore.

have a nice day.
